sklearn/model_selection/_search.py

```diff
@@ -965,11 +965,12 @@ def _store(key_name, array, weights=None, splits=False, rank=False):
             results["std_%s" % key_name] = array_stds
 
             if rank:
-                rank_result = rankdata(-array_means, method="min")
                 # when input is nan, scipy >= 1.10 rankdata returns nan. To
-                # keep previous behaviour nans are set to the
-                # maximum possible rank
-                rank_result[np.isnan(rank_result)] = len(rank_result)
+                # keep previous behaviour nans are set to be smaller than the
+                # minimum value in the array before ranking
+                min_array_means = min(array_means) - 1
+                np.nan_to_num(array_means, copy=False, nan=min_array_means)
```

Comment:
This comment is mostly for my education, because I don't know what the answer is.

With `copy=False` the `array_means` array is modified (all NaNs will be replaced by a number). I think this array is also accessible to users of the grid search as `cv_results_`. Which makes me wonder: do we need to tell users that the contents of this attribute has changed? Or make it so that the contents doesn't change (by using `copy=True`)?

Reply:
Good catch! Indeed it affects `self.cv_results_["mean_test_*"]`. Something like `[0.5, 0.5, np.nan]` gets mapped to `[0.5, 0.5, -0.5]`.

We can go for a small copy here. Alternatively, we could calculate the mask of `np.nan` in the array, temporarily swap them with `min(array_means) - 1` and substitute them back in later, but I think the copy is simpler. I don't think the array will ever be particularly huge, so this should be fine?
```suggestion
                array_means = np.nan_to_num(array_means, copy=True, nan=min_array_means)
```